lib/dirty/dirty.js

```diff
@@ -5,205 +5,199 @@ var fs = require('fs'),
     EventEmitter = require('events').EventEmitter;
 
 
-/**
-* Constructor function
-*/
-var Dirty = exports.Dirty = function(path) {
-  if (!(this instanceof Dirty)) return new Dirty(path);
+class Dirty extends EventEmitter {
+  constructor(path) {
```

Comment:
Right now users aren't required to use `new`, so this is not a backwards compatible change. Some options:
  * Bump major version.
  * Export a wrapper `function` around the class rather than the class itself.
  * Use a Proxy to trap `apply`:
    ```javascript
    // Trap `apply` for backwards compatibility with callers that don't use `new`.
    module.exports = exports = new Proxy(Dirty, {apply: (target, thisArg, args) => new Dirty(...args)});
    ```
I think I prefer the last option. What do you think?

Reply:
proxy seems like a good idea.
all doe... converting to class is also a breaking change in itself. Not syntax wise, but compatibility wise.

what if we make a new breaking change and also switch the the pkg from commonjs to ESM? That way you can import the pkg with node's new http loader also and don't have to use npm

Reply:
> proxy seems like a good idea.

Cool. I pushed a commit that I'll squash with the others.

> all doe... converting to class is also a breaking change in itself. Not syntax wise, but compatibility wise.

True, but it only breaks compatibility for users that are using very ancient, unsupported versions of Node.js. I don't think those users will be upgrading node-dirty any time soon. :slightly_smiling_face: 

> 
> what if we make a new breaking change and also switch the the pkg from commonjs to ESM? That way you can import the pkg with node's new http loader also and don't have to use npm

I haven't yet made the switch to ESM for anything I use, so I'm not familiar with the compatibility issues that might arise. I definitely don't want to switch in this PR, but maybe it would be OK in a separate PR.